shell/platform/android/external_view_embedder/external_view_embedder.cc

```diff
@@ -43,23 +58,142 @@ std::vector<SkCanvas*> AndroidExternalViewEmbedder::GetCurrentCanvases() {
   return canvases;
 }
 
+SkRect AndroidExternalViewEmbedder::GetViewRect(int view_id) {
+  EmbeddedViewParams* params = &view_params_[view_id];
+  FML_CHECK(params);
+  return SkRect::MakeXYWH(params->offsetPixels.x(),                          //
+                          params->offsetPixels.y(),                          //
+                          params->sizePoints.width() * device_pixel_ratio_,  //
+                          params->sizePoints.height() * device_pixel_ratio_  //
+  );
+}
+
 // |ExternalViewEmbedder|
 bool AndroidExternalViewEmbedder::SubmitFrame(
     GrContext* context,
     std::unique_ptr<SurfaceFrame> frame) {
-  // TODO(egarciad): Implement hybrid composition.
-  // https://github.com/flutter/flutter/issues/55270
   TRACE_EVENT0("flutter", "AndroidExternalViewEmbedder::SubmitFrame");
+
   if (should_run_rasterizer_on_platform_thread_) {
     // Don't submit the current frame if the frame will be resubmitted.
     return true;
   }
+
+  std::map<int64_t, std::list<SkRect>> overlay_layers;
+  std::map<int64_t, sk_sp<SkPicture>> pictures;
+  SkCanvas* background_canvas = frame->SkiaCanvas();
+
+  // Restore the clip context after exiting this method since it's changed
+  // below.
+  SkAutoCanvasRestore save(background_canvas, /*doSave=*/true);
+
   for (size_t i = 0; i < composition_order_.size(); i++) {
     int64_t view_id = composition_order_[i];
-    frame->SkiaCanvas()->drawPicture(
-        picture_recorders_[view_id]->finishRecordingAsPicture());
+    SkRect view_rect = GetViewRect(view_id);
+
+    // Display the platform view. If it's already displayed, then it's
+    // just positioned and sized.
+    jni_facade_->FlutterViewOnDisplayPlatformView(view_id,            //
+                                                  view_rect.x(),      //
+                                                  view_rect.y(),      //
+                                                  view_rect.width(),  //
+                                                  view_rect.height()  //
+    );
+
+    pictures[view_id] = picture_recorders_[view_id]->finishRecordingAsPicture();
```

Comment:
Check for null pictures when the recording is finished. Returning false from this method in case of any error is fine.

Reply:
Done.  I added `FML_CHECK` as based on https://github.com/google/skia/blob/e3986755dbb1e3d18793ec3508b4e247449b2196/include/core/SkPictureRecorder.h#L61 this is an incorrect state.